src/unix/stream.c

```diff
@@ -394,7 +394,8 @@ int uv__stream_open(uv_stream_t* stream, int fd, int flags) {
 #if defined(__APPLE__)
   enable = 1;
   if (setsockopt(fd, SOL_SOCKET, SO_OOBINLINE, &enable, sizeof(enable)) &&
-      errno != ENOTSOCK) {
+      errno != ENOTSOCK &&
+      errno != EINVAL) {
```

Comment:
There are socket types that don't support out-of-band data, so setting SO_OOBINLINE on those would fail with ENOPROTOOPT.
If someone uses uv_xxx_open it is deferred to uv__stream_open eventually, so basically any type of fd could be passed through this function.